app/src/main/kotlin/at/bitfire/davdroid/push/PushNotificationManager.kt

```diff
@@ -0,0 +1,64 @@
+package at.bitfire.davdroid.push
+
+import android.accounts.Account
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import at.bitfire.davdroid.R
+import at.bitfire.davdroid.ui.NotificationRegistry
+import at.bitfire.davdroid.ui.account.AccountActivity
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+class PushNotificationManager @Inject constructor(
+    @ApplicationContext private val context: Context,
+    private val notificationRegistry: NotificationRegistry
+) {
+
+    /**
+     * Generates the notification ID for a push notification.
+     */
+    private fun notificationId(account: Account, authority: String): Int {
+        return account.name.hashCode() + account.type.hashCode() + authority.hashCode()
+    }
+
+    /**
+     * Sends a notification to inform the user that a push notification has been received, the
+     * sync has been scheduled, but it still has not run.
+     */
+    fun notify(account: Account, authority: String) {
+        notificationRegistry.notifyIfPossible(notificationId(account, authority)) {
+            NotificationCompat.Builder(context, notificationRegistry.CHANNEL_STATUS)
+                .setSmallIcon(R.drawable.ic_sync)
+                .setContentTitle(context.getString(R.string.sync_notification_pending_push_title))
+                .setContentText(context.getString(R.string.sync_notification_pending_push_message))
+                .setSubText(account.name)
+                .setPriority(NotificationCompat.PRIORITY_LOW)
+                .setCategory(NotificationCompat.CATEGORY_STATUS)
+                .setAutoCancel(true)
+                .setOnlyAlertOnce(true)
+                .setContentIntent(
+                    PendingIntent.getActivity(
+                        context,
+                        0,
+                        Intent(context, AccountActivity::class.java).apply {
+                            putExtra(AccountActivity.EXTRA_ACCOUNT, account)
+                        },
+                        PendingIntent.FLAG_IMMUTABLE
```

Comment:
Do we need `PendingIntent.FLAG_UPDATE_CURRENT` or something like that? It would replace the extras for existing PendingIntents with the same target if I remember correctly.